project-metrics/metrics_service/test/metric_test_case.py

```diff
@@ -0,0 +1,72 @@
+"""Base class for testing metrics."""
+
+import abc
+import unittest
+from unittest import mock
+import sqlalchemy
+from sqlalchemy import orm
+
+from database import models
+from metrics import base
+
+
+class MetricTestCase(unittest.TestCase):
+
+  __metaclass__ = abc.ABCMeta
+
+  @classmethod
+  def setUpClass(cls):
+    super(MetricTestCase, cls).setUpClass()
+
+    cls.engine = sqlalchemy.create_engine('sqlite:///:memory:', echo=True)
```

Comment:
How heavy is it to create a new engine for every test (that is, in `setUp` as opposed to `setUpClass`) or to at least put the schema init/dropping in the `setUp` and `tearDown`?
Expecting the subclasses to know always do the tearDown correctly is a source of future misery for when we add a new metric 20 months from now and fail to figure out how to write tests then

Reply:
It approximately doubles the time to run those tests, but since it's doubling from 0.06s to 0.13s I feel like it's worth it :laughing: